discovery/legacymanager/manager_test.go

```diff
@@ -824,13 +824,9 @@ func TestTargetSetRecreatesEmptyStaticConfigs(t *testing.T) {
 	if !ok {
 		t.Fatalf("'%v' should be present in target groups", pkey)
 	}
-	group, ok := targetGroups[""]
-	if !ok {
-		t.Fatalf("missing '' key in target groups %v", targetGroups)
-	}
-
-	if len(group.Targets) != 0 {
```

Comment:
A small suggestion, you should not delete the original test, instead you should add a test for the situation you said to verify that what you said is correct.

Reply:
> A small suggestion, you should not delete the original test, instead you should add a test for the situation you said to verify that what you said is correct.

Thanks for your suggestion. I found the original test covers my tests，but it needs some modifications to pass.